### ISSUE TITLE
dangling semicolon after import causes UnsupportedOperationException

### ISSUE THREAD

**Issue by jojochuang** [javaparser/javaparser#487]
A "UnsupportedOperationException: Empty import declarations have no name" was thrown when it parsed the code that contains a dangling semicolon at the following line:

https://github.com/apache/hadoop/blob/2ca73445f5c2929d9c2ff4232dca58a63a0570a0/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/CacheReplicationMonitor.java#L59

Stacktrace:
Exception in thread "main" java.lang.UnsupportedOperationException: Empty import declarations have no name
    at com.github.javaparser.ast.ImportDeclaration.getName(ImportDeclaration.java:117)
    at com.github.javaparser.ast.visitor.VoidVisitorAdapter.visit(VoidVisitorAdapter.java:505)
    at com.github.javaparser.ast.ImportDeclaration.accept(ImportDeclaration.java:106)
    at com.github.javaparser.ast.visitor.VoidVisitorAdapter.visit(VoidVisitorAdapter.java:291)

The JavaParser version is 3.0-SNAPSHOT

**Comment by matozoid** [comment]
If I run the following code on the current master branch, there is no error. Can you make it fail?

``` java
package com.github.javaparser;

import com.github.javaparser.ast.CompilationUnit;
import org.junit.Test;

import java.io.IOException;
import java.net.URL;

public class Issue487 {
    @Test
    public void issue487() throws IOException {
        URL url = new URL("https://raw.githubusercontent.com/apache/hadoop/2ca73445f5c2929d9c2ff4232dca58a63a0570a0/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/CacheReplicationMonitor.java");
        CompilationUnit parsed = JavaParser.parse(url.openStream());
        System.out.println(parsed);
    }
}
```

**Comment by ptitjes** [comment]
Git blame says: https://github.com/javaparser/javaparser/commit/2751db301c03238088b5f2cec5e3c2f8aea6edc5#diff-7be095a97eb671ac3a092800247a7b35R105

**Comment by matozoid** [comment]
I vaguely remember the discussion back then. It seems fine to me to throw those exceptions there.

**Comment by matozoid** [comment]
Actually, it seems that some visitors don't take this into account, but that's another issue than is being reported here. Opened #488

**Comment by jojochuang** [comment]
@matozoid that was a really quick response. Thanks!
FYI the code I wrote is here https://github.com/jojochuang/codebetter/blob/master/src/main/java/LogMissingStackTrace.java and it uses VoidVisitorAdapter.

**Comment by matozoid** [comment]
Alright, so then it _is_ issue #488 - you can make a quick fix by overriding a method in your visitor:

``` java
    @Override
    public void visit(final ImportDeclaration n, final A arg) {
        visitComment(n.getComment(), arg);
        if (!n.isEmptyImportDeclaration()) {
            n.getName().accept(this, arg);
        }
    }
```

Thanks to @ptitjes for making me think about this ;)

I'd like to give the issue itself a little thought before fixing it.

**Comment by matozoid** [comment]
A definitive fix has been proposed in #491 and could use some feedback.

**Comment by matozoid** [comment]
Fix is now on master and will be in the next release.
